scripts/Muon/GUI/Common/pairing_table_widget/pairing_table_widget_presenter.py

```diff
@@ -227,7 +227,7 @@ def remove_last_row_in_view_and_model(self):
             name = self._view.get_table_contents()[-1][0]
             warning = self._model.check_pair_in_use(name)
             if warning:
-                self._View.warning_popup(warning)
```

Comment:
Is there a test we can add to catch this?

Reply:
Yes. I am in the process of writing tests here #30687. This PR will add a test to cover this. to make it easy for a patch the fix has been done separately